### PR TITLE
Support notifier's new parameter system.

### PR DESCRIPTION
**Pull request by ozmodiar**
This PR contains the MailChannel implementation of the new proposed parameter system (https://github.com/Notifier/Notifier/pull/19).

### CODE REVIEW

**Comment by ozmodiar** [comment]
Test will succeed when the notifier branch is merged...

File: src/MailChannel.php [hunk 1]
```diff
@@ -29,9 +39,7 @@ class MailChannel implements ChannelInterface
      */
     public function isHandling(MessageInterface $message, RecipientInterface $recipient)
     {
-        return isset($recipient->mail_to)
-            && isset($message->mail_subject)
-            && isset($message->mail_body);
+        return ($this->getMessageBag($message) && $this->getRecipientBag($recipient));
```
Comment (NoUseFreak) [inline src/MailChannel.php]: Remove the parenthesis. Maybe do ! `is_null` checks when the bags return `null`.

**Comment by ozmodiar** [comment]
Changed a few things here and there. Let me know if I can rebase this...

File: tests/Stubs/ParameterBag.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        return $this->identifier;
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param string $identifier
+     */
+    public function __construct($identifier)
```
Comment (NoUseFreak) [inline tests/Stubs/ParameterBag.php:29]: Place constructors as first method.

File: src/ParameterBag/MailRecipientParameterBag.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * @author Joeri van Dooren
+ */
+class MailRecipientParameterBag implements ParameterBagInterface
+{
+    /**
+     * @var string
+     */
+    private $to;
```
Comment (NoUseFreak) [inline src/ParameterBag/MailRecipientParameterBag.php:26]: Can you indicate what the function of $to is. Indicating it is an email will make the implementation more clear.

File: src/MailChannel.php [hunk 4]
```diff
@@ -43,16 +58,41 @@ public function isHandling(MessageInterface $message, RecipientInterface $recipi
...
+            $messageBag->getHeaders(),
+            $messageBag->getParameters()
         );
     }
 
     /**
+     * Get the MailParameterBag from the recipient.
+     *
+     * @param \Notifier\Recipient\RecipientInterface $recipient
+     * @return bool|MailRecipientParameterBag
```
Comment (NoUseFreak) [inline src/MailChannel.php]: Add a newline before the return statement.

File: src/MailChannel.php [hunk 5]
```diff
@@ -43,16 +58,41 @@ public function isHandling(MessageInterface $message, RecipientInterface $recipi
...
+    protected function getRecipientBag(RecipientInterface $recipient)
+    {
+        return $recipient->getParameterBag($this->getIdentifier());
+    }
+
+    /**
+     * Get the MailParameterBag from the message.
+     *
+     * @param \Notifier\Message\MessageInterface $message
+     * @return bool|MailMessageParameterBag
```
Comment (NoUseFreak) [inline src/MailChannel.php]: Add a newline before the return statement.

Comment (NoUseFreak) [inline src/MailChannel.php]: The return statement should return only one type. A bag or null.

File: src/MailChannel.php (hunk 4 above)
Comment (NoUseFreak) [inline src/MailChannel.php]: The return statement should return only one type. A bag or null.

File: tests/Stubs/ParameterBag.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        $this->identifier = $identifier;
+    }
+}
```
Comment (NoUseFreak) [inline tests/Stubs/ParameterBag.php:41]: Add newlines after every file.

File: src/ParameterBag/MailRecipientParameterBag.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    /**
+     * @return string
+     */
+    public function getIdentifier()
+    {
+        return MailChannel::IDENTIFIER;
+    }
+}
```
Comment (NoUseFreak) [inline src/ParameterBag/MailRecipientParameterBag.php:53]: Add newlines after every file.

File: src/ParameterBag/MailMessageParameterBag.php [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    /**
+     * @return string
+     */
+    public function getIdentifier()
+    {
+        return MailChannel::IDENTIFIER;
+    }
+}
```
Comment (NoUseFreak) [inline src/ParameterBag/MailMessageParameterBag.php:96]: Add newlines after every file.

**Comment by ozmodiar** [comment]
Code style fixed done. Can you check the empty lines at the end of files before commenting? They were all there :-)

**Comment by NoUseFreak** [comment]
> Can you check the empty lines at the end of files before commenting? They were all there :-)

It looks different on the [file previews](https://github.com/Ozmodiar/Mail/blob/parameterbag-support/src/ParameterBag/MailRecipientParameterBag.php). If they are present you can ignore the remark. :-)

**Pull request merged**